test/libsolidity/syntaxTests/parsing/errorRecovery/sync_to_semicolon.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity >=0.0.0;
+
+// This is a contract with a number of syntax errors in
+// statements where deleting to semicolon seems to be
+// the right thing to do
+
+// Tokens are deleted until the next subsequent semicolon.
+
+contract MetaCoin2 {
+	mapping (address => uint) balances;
+
+	event Transfer(address indexed _from, address indexed _to, uint256 _value);
+
+	constructor() public {
+		balances[tx.origin] = 10000;
+	}
+
+	function sendCoin(address receiver, uint amount) public returns(bool sufficient) {
+		uint i;
+		if (balances[msg.sender])
+		    return false;
+
+		balances[msg.sender] -= amount;
+
+		balances[receiver] += amount;
+		emit Transfer(msg.sender, receiver,);  // missing param or extra ,
+
+		// missing ) in while
+		while (amount < 10) {
+			if (amount > 1) break;
+				amount += ;  //
+			continue;
+			if (amount > 2) break;
+		}
+		return true;
+	}
+
+	function getBalanceInEth(address addr) public view returns(uint){
+		return ConvertLib.convert(getBalance(addr),2);
+	}
+
+	function getBalance(address addr) public view returns(uint) {
+		return balances[addr];
+	}
+}
+// ----
+// ParserError: (530-538): Expected ';' but got identifier. Skipping to next ';'.
```

Comment:
The specific errors are not right. Nor is this complete. This is just a template of what I am using right now and will later contain more errors.